### PR TITLE
Adding a before publishing traverse to avoid the default view on REST calls

### PR DESCRIPTION
**Pull request by tisto**

### CODE REVIEW

**Comment by tisto** [comment]
http://jenkins.plone.org/job/pull-request-5.0/82/

File: plone/dexterity/content.py [hunk 1]
```diff
@@ -671,6 +695,21 @@ class Container(
     Description = DexterityContent.Description
     setDescription = DexterityContent.setDescription
 
+    def __before_publishing_traverse__(self, arg1, arg2=None):
```
Comment (tisto) [inline plone/dexterity/content.py]: @bloodbare __before_publishing_traverse__ is listed twice here. Merge problem?

**Comment by tisto** [comment]
@bloodbare the jenkins pull request job fails with 91 test failures:

http://jenkins.plone.org/job/pull-request-5.0/82/testReport/

Would you mind having a look?

**Comment by jensens** [comment]
this is a massive new behavior, where is the PLIP?

**Comment by lukasgraf** [comment]
These changes lead to the breakage outlined in plone/plone.rest#14 (first TB). This is because the `__before_publishing_traverse__` for `Container` actually does a super call for `super(DexterityContent, self)`.

With the major cleanup of the monkey patches in plone/plone.rest#21 I really question whether this change is still warranted.

Also, there seems to be a PLIP placeholder at plone/Products.CMFPlone#680, but no information in it whatsoever. What's the scope of that PLIP?

@tisto @bloodbare what's the status here? My suggestion would be to close this, [switch the `plone.restapi` buildout back to `plone.dexterity:master`](https://github.com/plone/plone.restapi/blob/17b585a4c76704a0a6f89524f8a87792c519ddfe/buildout.cfg#L87) and go with the simplified monkey patch in `plone.restapi` for now. If there's need, we can revisit integrating these changes into `plone.dexterity` at a later point.

**Comment by jensens** [comment]
This Pull Request has stalled for a while now and is closed. If you care about this Pull Request please reopen at any time.

**Pull request closed without merging**